### DISCUSSION TITLE
Add support for Azure Pipelines' new deprecated metadata on tasks

### DISCUSSION THREAD

**Discussion by jessehouwing** [renovatebot/renovate#25498] [Suggest an Idea]
### Tell us more.

Azure Pipelines has added metadata on tasks to mark them deprecated. It will contain a boolean and a kill-date.

Pipelines will issue the warning every time the pipeline runs:

```
If you're using the NuGet Restore v1 and NuGet Installer v0 pipeline tasks, promptly transition to the NuGetCommand@2 pipeline task. You'll begin receiving alerts in your pipelines soon if the transition hasn't been made. If no action is taken, starting November 27, 2023, your builds will result in failure.
```

The metadata is in the `task.json`:

```
  "deprecated": true,
  "removalDate": "2023-11-27",
```

https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/NuGetInstallerV0/task.json#L20-L21

This would be good to detect by renovate and log an error/warning against.

The API already returns `deprecated: true`, but unfortunately doesn't return the `removalDate`. I've raised this with the Azure Pipelines Team. The message is currently hard-coded in the `.js` file of the task and not easily retrieved.

**Comment by secustor** [comment]
I'm in favour of this.

Some thoughts of mine. There are two things here:
- Getting the deprecation status in Renovate
  - add this metadata to https://github.com/renovatebot/azure-devops-marketplace
  - extend the `azure-pipelines-tasks` to read the deprecation status
- Logging that the current version has been deprecated. ( maybe Dashboard entry? )

**Reply by jessehouwing** [reply]
It should be relatively simple to update the generated Json file. Ideally if there is an alternative, a rename rule could be added as well.

As of yet the data isn't available in the REST api yet. I've asked the Azure DevOps team if that could be added.
